File: Anzeige2/src/main/java/com/capgemini/capybench/Anzeige2/entity/Topic.java
package com.capgemini.capybench.Anzeige2.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.Objects;
import java.util.Set;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter
@Setter
public class Topic {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column
    private String name;

    @Column
    private LocalDateTime expirationDate;

    // Subscribers
    @ManyToMany(
            mappedBy = "subscribedTopics",
            fetch = FetchType.LAZY)
    private Set<Person> people;

    @OneToMany(
            mappedBy = "topic",
            fetch = FetchType.LAZY)
    private Set<Subtopic> subtopics;

    @OneToMany(
            mappedBy = "topic",
            fetch = FetchType.LAZY)
    private Set<Post> posts;

    public void addSubscriber(Person person){
        people.add(person);
    }

    @Override
    public boolean equals(final Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        Topic topic = (Topic) o;
        return Objects.equals(id, topic.id) && Objects.equals(name, topic.name);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, name);
    }

}

File: Anzeige 2.0/src/domain/Topic.ts
import { Subtopic } from "./Subtopic";

export class Topic {
    id: number;
    name: string;
    subtopics: Subtopic[];
    postIds: number[];
    subscriberIds: number[];
    expirationDate: Date;

    constructor(id: number, name: string, subtopics: Subtopic[], postIds: number[], subsciberIds: number[], expirationDate: Date) {
        this.id = id;
        this.name = name;
        this.subtopics = subtopics;
        this.postIds = postIds;
        this.subscriberIds = subsciberIds;
        this.expirationDate = expirationDate;
    }
}